### DISCUSSION TITLE
Option to use tab to autocomplete query

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#516] [Ideas]
From https://github.com/jjw24/Wox/pull/153
![image](https://user-images.githubusercontent.com/26427004/93663957-be2b8c80-faae-11ea-881f-55914d0ecefe.png)

**Comment by jjw24** [comment]
@taooceros move back to issues? this is ok to proceed to implement

**Reply by taooceros** [reply]
Currently `tab` is used for iterate to the next item. I didn't see a lot of usage for autocomplete to be used, so maybe we can discuss about it.

**Reply by jjw24** [reply]
ok no problem

**Comment by jjw24** [comment]
one usage for this i think, will be for shell plugin, to auto complete commands eg, there is a result already for 'ping google.com', i can with shell command type 'ping goo' and tab to autocomplete it

**Reply by pc223** [reply]
Users can use Ctrl+N for Next and Tab for autocomplete, similar to linux/bash/vim, I think it's really interesting to try. 

Not sure how many users accept to move from Tab to Ctrl +N tho 😅 , For now we can add a opt-in setting "Tab to autocomplete instead of next item"

> . I didn't see a lot of usage for autocomplete to be used

@taooceros 
How to do tab-to-autocomplete in Flow? I never knew we can do it

**Reply by taooceros** [reply]
I think propsal is willing to change the query to Result.Title?

**Reply by pc223** [reply]
> I think propsal is willing to change the query to Result.Title?

Hm...I would think so yeah, I think we should focus polish the UX on Explorer and Shell, these two really benefit from autocomplete, other plugins, not so much, at least for now.

**Reply by jjw24** [reply]
> I think propsal is willing to change the query to Result.Title?

yeah at this stage i can think we can just have an option in settings to use tab as autocomplete instead of next, then when tab is pressed the current top result's title where it starts with the current query typed, then change the query to the result title. then cycle through the list of results if you keep tabbing.
